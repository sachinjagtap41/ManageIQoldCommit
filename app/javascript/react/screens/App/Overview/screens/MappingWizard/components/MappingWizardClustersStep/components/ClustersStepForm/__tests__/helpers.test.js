import {
  targetClusterWithExtendedData,
  sourceClusterWithExtendedData,
  updateMapping,
  createNewMapping
} from '../helpers';

import { srcClusters, tgtClusters } from '../clustersStepForm.fixtures';

test('targetClusterWithExtendedData adds attrs for displaying in TreeView', () => {
  const [targetCluster] = tgtClusters;

  expect(targetClusterWithExtendedData(targetCluster)).toMatchSnapshot();
});

test('sourceClusterWithExtendedData adds attrs for displaying in TreeView', () => {
  const [sourceCluster] = srcClusters;

  expect(sourceClusterWithExtendedData(sourceCluster)).toMatchSnapshot();
});

describe('updateMapping', () => {
  test('adds source clusters to mapping if target cluster id matches', () => {
    const [mappedTargetCluster] = tgtClusters;
    const [mappedSourceCluster, sourceClusterToAdd] = srcClusters;
    const clustersStepMapping = {
      ...mappedTargetCluster,
      nodes: [mappedSourceCluster]
    };

    expect(
      updateMapping(clustersStepMapping, mappedTargetCluster, [
        sourceClusterToAdd
      ])
    ).toMatchSnapshot();
  });

  test('does not add source clusters if target cluster id does not match', () => {
    const [mappedTargetCluster, targetClusterToAddTo] = tgtClusters;
    const [mappedSourceCluster, sourceClusterShouldNotBeAdded] = srcClusters;
    const clustersStepMapping = {
      ...mappedTargetCluster,
      nodes: [mappedSourceCluster]
    };

    expect(
      updateMapping(clustersStepMapping, targetClusterToAddTo, [
        sourceClusterShouldNotBeAdded
      ])
    ).toMatchSnapshot();
  });
});

test('createNewMapping constructs a clusters step mapping', () => {
  const [targetCluster] = tgtClusters;
  const [sourceCluster] = srcClusters;

  expect(createNewMapping(targetCluster, [sourceCluster])).toMatchSnapshot();
});
