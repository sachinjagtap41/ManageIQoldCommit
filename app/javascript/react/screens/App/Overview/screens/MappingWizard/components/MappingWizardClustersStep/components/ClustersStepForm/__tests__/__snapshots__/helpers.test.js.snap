// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createNewMapping constructs a clusters step mapping 1`] = `
Object {
  "created_on": "2016-08-03T02:19:35Z",
  "drs_automation_level": "fullyAutomated",
  "drs_enabled": false,
  "drs_migration_threshold": 3,
  "effective_cpu": 109356,
  "effective_memory": 789977235456,
  "ems_id": "10000000000007",
  "ems_ref": "domain-c7",
  "ems_ref_obj": "domain-c7",
  "ext_management_system": Object {
    "emstype": "rhevm",
  },
  "ha_admit_control": true,
  "ha_enabled": false,
  "ha_max_failures": 1,
  "href": "http://localhost:3000/api/clusters/10000000000006",
  "id": "1",
  "last_perf_capture_on": null,
  "name": "RHV Cluster1",
  "nodes": Array [
    Object {
      "created_on": "2016-07-15T18:28:16Z",
      "drs_automation_level": null,
      "drs_enabled": null,
      "drs_migration_threshold": null,
      "effective_cpu": null,
      "effective_memory": null,
      "ems_id": "10000000000012",
      "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ext_management_system": Object {
        "emstype": "vmwarews",
      },
      "ha_admit_control": null,
      "ha_enabled": null,
      "ha_max_failures": null,
      "href": "http://localhost:3000/api/clusters/10000000000001",
      "icon": "fa fa-file-o",
      "id": "1",
      "last_perf_capture_on": null,
      "name": "VMWare Cluster1",
      "text": "VMWare Cluster1",
      "type": null,
      "uid_ems": "00000002-0002-0002-0002-00000000024b",
      "updated_on": "2016-07-15T18:28:16Z",
      "v_parent_datacenter": "Datacenter",
    },
  ],
  "selectable": true,
  "selected": false,
  "state": Object {
    "expanded": true,
  },
  "text": "RHV Cluster1",
  "type": null,
  "uid_ems": "domain-c7",
  "updated_on": "2017-02-26T02:33:43Z",
  "v_parent_datacenter": "Default",
}
`;

exports[`sourceClusterWithExtendedData adds attrs for displaying in TreeView 1`] = `
Object {
  "created_on": "2016-07-15T18:28:16Z",
  "drs_automation_level": null,
  "drs_enabled": null,
  "drs_migration_threshold": null,
  "effective_cpu": null,
  "effective_memory": null,
  "ems_id": "10000000000012",
  "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
  "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
  "ext_management_system": Object {
    "emstype": "vmwarews",
  },
  "ha_admit_control": null,
  "ha_enabled": null,
  "ha_max_failures": null,
  "href": "http://localhost:3000/api/clusters/10000000000001",
  "icon": "fa fa-file-o",
  "id": "1",
  "last_perf_capture_on": null,
  "name": "VMWare Cluster1",
  "text": "VMWare Cluster1",
  "type": null,
  "uid_ems": "00000002-0002-0002-0002-00000000024b",
  "updated_on": "2016-07-15T18:28:16Z",
  "v_parent_datacenter": "Datacenter",
}
`;

exports[`targetClusterWithExtendedData adds attrs for displaying in TreeView 1`] = `
Object {
  "created_on": "2016-08-03T02:19:35Z",
  "drs_automation_level": "fullyAutomated",
  "drs_enabled": false,
  "drs_migration_threshold": 3,
  "effective_cpu": 109356,
  "effective_memory": 789977235456,
  "ems_id": "10000000000007",
  "ems_ref": "domain-c7",
  "ems_ref_obj": "domain-c7",
  "ext_management_system": Object {
    "emstype": "rhevm",
  },
  "ha_admit_control": true,
  "ha_enabled": false,
  "ha_max_failures": 1,
  "href": "http://localhost:3000/api/clusters/10000000000006",
  "id": "1",
  "last_perf_capture_on": null,
  "name": "RHV Cluster1",
  "selectable": true,
  "selected": false,
  "state": Object {
    "expanded": true,
  },
  "text": "RHV Cluster1",
  "type": null,
  "uid_ems": "domain-c7",
  "updated_on": "2017-02-26T02:33:43Z",
  "v_parent_datacenter": "Default",
}
`;

exports[`updateMapping adds source clusters to mapping if target cluster id matches 1`] = `
Object {
  "created_on": "2016-08-03T02:19:35Z",
  "drs_automation_level": "fullyAutomated",
  "drs_enabled": false,
  "drs_migration_threshold": 3,
  "effective_cpu": 109356,
  "effective_memory": 789977235456,
  "ems_id": "10000000000007",
  "ems_ref": "domain-c7",
  "ems_ref_obj": "domain-c7",
  "ext_management_system": Object {
    "emstype": "rhevm",
  },
  "ha_admit_control": true,
  "ha_enabled": false,
  "ha_max_failures": 1,
  "href": "http://localhost:3000/api/clusters/10000000000006",
  "id": "1",
  "last_perf_capture_on": null,
  "name": "RHV Cluster1",
  "nodes": Array [
    Object {
      "created_on": "2016-07-15T18:28:16Z",
      "drs_automation_level": null,
      "drs_enabled": null,
      "drs_migration_threshold": null,
      "effective_cpu": null,
      "effective_memory": null,
      "ems_id": "10000000000012",
      "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ext_management_system": Object {
        "emstype": "vmwarews",
      },
      "ha_admit_control": null,
      "ha_enabled": null,
      "ha_max_failures": null,
      "href": "http://localhost:3000/api/clusters/10000000000001",
      "id": "1",
      "last_perf_capture_on": null,
      "name": "VMWare Cluster1",
      "type": null,
      "uid_ems": "00000002-0002-0002-0002-00000000024b",
      "updated_on": "2016-07-15T18:28:16Z",
      "v_parent_datacenter": "Datacenter",
    },
    Object {
      "created_on": "2016-07-15T18:28:16Z",
      "drs_automation_level": null,
      "drs_enabled": null,
      "drs_migration_threshold": null,
      "effective_cpu": null,
      "effective_memory": null,
      "ems_id": "10000000000012",
      "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ext_management_system": Object {
        "emstype": "vmwarews",
      },
      "ha_admit_control": null,
      "ha_enabled": null,
      "ha_max_failures": null,
      "href": "http://localhost:3000/api/clusters/10000000000002",
      "icon": "fa fa-file-o",
      "id": "2",
      "last_perf_capture_on": null,
      "name": "VMWare Cluster2",
      "text": "VMWare Cluster2",
      "type": null,
      "uid_ems": "00000002-0002-0002-0002-00000000024b",
      "updated_on": "2016-07-15T18:28:16Z",
      "v_parent_datacenter": "Datacenter",
    },
  ],
  "type": null,
  "uid_ems": "domain-c7",
  "updated_on": "2017-02-26T02:33:43Z",
  "v_parent_datacenter": "Default",
}
`;

exports[`updateMapping does not add source clusters if target cluster id does not match 1`] = `
Object {
  "created_on": "2016-08-03T02:19:35Z",
  "drs_automation_level": "fullyAutomated",
  "drs_enabled": false,
  "drs_migration_threshold": 3,
  "effective_cpu": 109356,
  "effective_memory": 789977235456,
  "ems_id": "10000000000007",
  "ems_ref": "domain-c7",
  "ems_ref_obj": "domain-c7",
  "ext_management_system": Object {
    "emstype": "rhevm",
  },
  "ha_admit_control": true,
  "ha_enabled": false,
  "ha_max_failures": 1,
  "href": "http://localhost:3000/api/clusters/10000000000006",
  "id": "1",
  "last_perf_capture_on": null,
  "name": "RHV Cluster1",
  "nodes": Array [
    Object {
      "created_on": "2016-07-15T18:28:16Z",
      "drs_automation_level": null,
      "drs_enabled": null,
      "drs_migration_threshold": null,
      "effective_cpu": null,
      "effective_memory": null,
      "ems_id": "10000000000012",
      "ems_ref": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ems_ref_obj": "/api/clusters/00000002-0002-0002-0002-00000000024b",
      "ext_management_system": Object {
        "emstype": "vmwarews",
      },
      "ha_admit_control": null,
      "ha_enabled": null,
      "ha_max_failures": null,
      "href": "http://localhost:3000/api/clusters/10000000000001",
      "id": "1",
      "last_perf_capture_on": null,
      "name": "VMWare Cluster1",
      "type": null,
      "uid_ems": "00000002-0002-0002-0002-00000000024b",
      "updated_on": "2016-07-15T18:28:16Z",
      "v_parent_datacenter": "Datacenter",
    },
  ],
  "type": null,
  "uid_ems": "domain-c7",
  "updated_on": "2017-02-26T02:33:43Z",
  "v_parent_datacenter": "Default",
}
`;
